src/compiler/performance.ts

```diff
@@ -107,6 +107,28 @@ namespace ts.performance {
         durations.forEach((duration, measureName) => cb(measureName, duration));
     }
 
+    export function forEachMark(cb: (markName: string) => void) {
+        marks.forEach((_time, markName) => cb(markName));
+    }
+
+    export function clearMeasures(name?: string) {
+        if (name) durations.delete(name);

```

Comment:
Not that we're likely to create a mark or measure named `""`, but to be consistent I'd recommend this:

```suggestion
        if (name === undefined) durations.delete(name);
```